testsuite/tests/conftest.py

```diff
@@ -183,6 +184,27 @@ def pytest_collection_modifyitems(session, config, items):
             item.user_properties.append(("issue", issue))
             item.user_properties.append(("issue-id", issue_id))
 
+    sso_only_opt = config.option.sso_only
+    if not sso_only_opt:
+        return
+
+    selected_fixtures = ["rhsso_setup"]
+    selected = list(items)
+    deselected = []
+
+    for item in items:
+        to_remove = True
+        for fixture in selected_fixtures:
+            if fixture in item.fixturenames:
+                to_remove = False
+
+        if to_remove:
+            selected.remove(item)
+            deselected.append(item)
+
+    items[:] = selected
```

Comment:
Cannot be simplified to ?
`   selected_fixtures = {"rhsso_setup"}
    deselected = []

    sso_only_opt = config.option.sso_only
    selected = []

    for item in items:
        for marker in item.iter_markers(name="issue"):
            issue = marker.args[0]
            issue_id = issue.rstrip("/").split("/")[-1]
            item.user_properties.append(("issue", issue))
            item.user_properties.append(("issue-id", issue_id))

        if sso_only_opt and not any(fixture in selected_fixtures for fixture in item.fixturenames):
            deselected.append(item)
        else:
            selected.append(item)

    items[:] = selected

    if deselected:
        config.hook.pytest_deselected(items=deselected)`